File: Backend/Travle.Model/Responses/TourDestinationRef.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// One ordered stop of a tour, as shown on the tour detail screen: the destination's name and city,
    /// its coordinates (so the stops can be mapped) and a small thumbnail. The ordering key
    /// (<see cref="SortOrder"/>) is preserved so the itinerary renders in the organizer's intended order.
    /// Full image bytes still come from the destination image endpoint — only the thumbnail travels here.
    /// </summary>
    public class TourDestinationRef
    {
        public int DestinationId { get; set; }

        public string Name { get; set; } = string.Empty;
        public string? CityName { get; set; }

        public double Latitude { get; set; }
        public double Longitude { get; set; }

        public int SortOrder { get; set; }

        public byte[]? Thumbnail { get; set; }
        public string? ThumbnailContentType { get; set; }
    }
}

File: UI/travle_core/lib/src/models/tour_destination_ref.dart
import 'package:json_annotation/json_annotation.dart';

part 'tour_destination_ref.g.dart';

/// One ordered stop of a tour (mirrors the backend `TourDestinationRef`): the
/// destination's name and city, its coordinates (so stops can be mapped) and a
/// small [thumbnail]. [sortOrder] preserves the itinerary order. Full image bytes
/// still come from the destination image endpoint — only the thumbnail is here.
@JsonSerializable()
class TourDestinationRef {
  TourDestinationRef({
    required this.destinationId,
    required this.name,
    required this.latitude,
    required this.longitude,
    required this.sortOrder,
    this.cityName,
    this.thumbnail,
    this.thumbnailContentType,
  });

  final int destinationId;
  final String name;
  final String? cityName;

  final double latitude;
  final double longitude;

  final int sortOrder;

  /// Base64 thumbnail bytes (null when the destination has no image).
  final String? thumbnail;
  final String? thumbnailContentType;

  factory TourDestinationRef.fromJson(Map<String, dynamic> json) =>
      _$TourDestinationRefFromJson(json);

  Map<String, dynamic> toJson() => _$TourDestinationRefToJson(this);
}
